src/test/java/com/egov/business-services/pretest/egfMasterAccountDetailsTypesPreTest.feature

```diff
@@ -37,4 +37,100 @@ Background:
     When method post
     Then status 400
     And def accountDetailTypesCreateResponseHeader = responseHeaders
-    And def accountDetailTypesCreateResponseBody = response
\ No newline at end of file
+    And def accountDetailTypesCreateResponseBody = response
+
+    @updateAccountSuccessfully
+  Scenario: Update Unique account detail type through API call
+    * def accountUpdateParam = 
+    """
+    {
+     tenantId: '#(tenantId)'
+    }
+    """ 
+    Given url accountDetailTypesUpdate
+    And params accountUpdateParam
+    And request accountDetailTypesUpdatePayload
+    When method post
+    Then status 201
+    And def accountDetailTypesUpdateResponseHeader = responseHeaders
+    And def accountDetailTypesUpdateResponseBody = response
+
+    @negativeUpdate
+  Scenario: Update Unique account detail type through API call
+    * def accountUpdateParam = 
+    """
+    {
+     tenantId: '#(tenantId)'
+    }
+    """ 
+    Given url accountDetailTypesUpdate
+    And params accountUpdateParam
+    And request accountDetailTypesUpdatePayload
+    When method post
+    Then status 400
+    And def accountDetailTypesUpdateResponseHeader = responseHeaders
+    And def accountDetailTypesUpdateResponseBody = response
+
+    @invalidTenantIdUpdate
+    Scenario: Invalid TenantID
+    * def accountUpdateParam = 
+    """
+    {
+     tenantId: '#(tenantId)'
+    }
+    """ 
+    Given url accountDetailTypesUpdate
+    And params accountUpdateParam
+    And request accountDetailTypesUpdatePayload
+    When method post
+    Then status 403
+    And def accountDetailTypesUpdateResponseHeader = responseHeaders
+    And def accountDetailTypesUpdateResponseBody = response
+
+    @searchAccount
+  Scenario: Search Unique account detail type through API call
+    * def accountSearchParam = 
+    """
+    {
+     tenantId: '#(tenantId)'
+    }
+    """ 
+    Given url accountDetailTypesSearch
+    And params accountSearchParam
+    And request accountDetailTypesSearchPayload
+    When method post
+    Then status 200
+    And def accountDetailTypesSearchResponseHeader = responseHeaders
+    And def accountDetailTypesSearchResponseBody = response
+
+     @negativeSearch
+  Scenario: Search Unique account detail type through API call
+    * def accountSearchParam = 
+    """
+    {
+     tenantId: '#(tenantId)'
+    }
+    """ 
+    Given url accountDetailTypesSearch
+    And params accountSearchParam
+    And request accountDetailTypesSearchPayload
+    When method post
+    Then status 403
+    And def accountDetailTypesSearchResponseHeader = responseHeaders
+    And def accountDetailTypesSearchResponseBody = response
+
```

Comment:
Please try to merge all `negative` pretests in one single pretest scenario, if the steps are similar for all. To validate the range of error status code, please use the below line.
```
assert responseStatus >=400 && responseStatus <=403 
```

Reply:
Review comments addressed. Please process.